sklearn/base.py

```diff
@@ -292,6 +293,24 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        if type(self).__module__.startswith('sklearn.'):
+            return dict(self.__dict__.items(), __version__=__version__)
+        else:
+            return dict(self.__dict__.items())
```

Comment:
> I think the class attribute issue is the bigger problem here. It doesn't get pickled unless it's set on the instance, does it?

Hum, you might be right. Darn, that may break my plan of simplicity.

Overriding a __ method always worries me, so I'd like to try to avoid, if
possible (the reason is that it gives non-standard behavior to objects,
hence more difficult comprehension of a non library, and more difficult
debuging.